test/ut/tools/nnictl/test_kill_command.py

```diff
@@ -0,0 +1,121 @@
+import argparse
+import multiprocessing
+import os
+import subprocess
+import signal
+import sys
+import signal
+import time
+
+import pytest
+
+from nni.tools.nnictl.command_utils import kill_command, _check_pid_running
+
+
+def process_normal():
+    time.sleep(360)
+
+
+def process_kill_slow(kill_time=2):
+    def handler_stop_signals(signum, frame):
+        print('debug proceess kill: signal received')
+        time.sleep(kill_time)
+        print('debug proceess kill: signal processed')
+        sys.exit(0)
+
+    signal.signal(signal.SIGINT, handler_stop_signals)
+    signal.signal(signal.SIGTERM, handler_stop_signals)
+    print('debug process kill: sleep')
+    time.sleep(360)
+
+
+def process_patiently_kill():
+    process = subprocess.Popen([sys.executable, __file__, '--mode', 'kill_very_slow'])
+    time.sleep(1)
+    kill_command(process.pid)  # wait long enough
+
+
+def test_kill_process():
+    process = multiprocessing.Process(target=process_normal)
+    process.start()
+
+    start_time = time.time()
+    kill_command(process.pid)
+    end_time = time.time()
+    assert not _check_pid_running(process.pid)
+    assert end_time - start_time < 2
+
+
+def test_kill_process_slow_no_patience():
+    process = subprocess.Popen([sys.executable, __file__, '--mode', 'kill_slow'])
+    time.sleep(1)  # wait 1 second for the process to launch and register hooks
+    start_time = time.time()
+    kill_command(process.pid, timeout=1)  # didn't wait long enough
+    end_time = time.time()
+    assert 0.5 < end_time - start_time < 2
+    if sys.platform == 'linux':  # FIXME: on non-linux, seems that the time of termination can't be controlled
```

Comment:
does this mean the handler in `signal.signal` will not register successfully on non-linux?

Reply:
I tested this on pipeline. Sometimes, the assert is successful. In other times, it's not. It's very strange.

I don't think it's a critical problem now: IT didn't complain; developers seldom use Windows.